From ced16e527cb61bdcaa177d8a29ec13fbe709c4b2 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Thu, 4 May 2023 17:20:10 +0200
Subject: [PATCH 0618/1695] plat-stm32mp2: update clock driver flag for
 STM32MP21

Add CFG_STM32MP21_CLK in conf.mk file that will be set when CFG_STM32MP21
is set. Else, set CFG_STM32MP25_CLK flag.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git b3f21aeff178a70d7aa53cc92d8532e0b37e003b plat-stm32mp2: update clock driver flag for STM32MP21]
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Signed-off-by: Nicolas Le Bayon <nicolas.le.bayon@st.com>
Change-Id: Id18927745a45680d97f8785e8d7487074c321490
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/361568
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
---
 core/arch/arm/plat-stm32mp2/conf.mk | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 8c6496f42..ad5162762 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -86,11 +86,16 @@ $(call force,CFG_STM32_PWR,y)
 $(call force,CFG_STM32_PWR_REGUL,y)
 $(call force,CFG_STM32MP_CLK_CORE,y)
 $(call force,CFG_STM32MP_REMOTEPROC,y)
-$(call force,CFG_STM32MP25_CLK,y)
 $(call force,CFG_STM32MP25_RSTCTRL,y)
 $(call force,CFG_WITH_ARM_TRUSTED_FW,y)
 $(call force,CFG_WITH_LPAE,y)
 
+ifeq ($(CFG_STM32MP21),y)
+$(call force,CFG_STM32MP21_CLK,y)
+else
+$(call force,CFG_STM32MP25_CLK,y)
+endif
+
 CFG_TZDRAM_START ?= 0x82000000
 CFG_TZDRAM_SIZE  ?= 0x02000000
 
-- 
2.47.2

